README.md

```diff
@@ -1,14 +1,105 @@
+saritasa/saritasa-tekton-apps     	0.1.20       	v0.28.2    	A Helm chart for tekton apps (rbac, eventlisten...
+saritasa/saritasa-tekton-pipelines	0.1.26       	           	A Helm chart for Tekton Pipelines  Implements: ...
+saritasa/demo                     	0.0.5        	1.16.0     	A Helm chart for Kubernetes  
+saritasa/jitsi-meet               	1.2.2        	stable-6865	A Helm chart for Kubernetes  
+saritasa/opsgenie-heartbeat       	0.0.2        	           	Chart that installs our solution to send heartb...
+```
+
+# Available helm charts
+
+## saritasa-rbac
+
+Chart providers rbac rules inside Kubernetes clusters
+
+## saritasa-tekton
+
+Chart provides tekton engine including:
+- dashboard
+- tektone engine itself
+- tekton triggers
+- tekton interceptors
+
+## saritasa-tekton-apps
+
+Chart provides tekton apps to configure tekton eventlistener
+
+## saritasa-tekton-pipelines
+
+Chart providers various pipelines for tekton
 ## demo
 
 For experiments - use anyway you want
 
-## jitsi-helm
+## jitsi-meet
+
+Chart providers WebRTC chat used in VP
+
+## opsgenie-heartbeat
+
+Chart providers opsgenie heartbeat for health notifications
+
+# Development
+
+## Required tools
+
+### go
+
+You will need to have go 1.18x installed on your local PC.
+
+```sh
+go version go1.18.4 linux/amd64
+```
+
+### helm-docs
+
+This is needed to generate helm docs automatically
+
+```sh
+GO111MODULE=on go get github.com/norwoodj/helm-docs/cmd/helm-docs
+```
+
+Your `values.yaml` file should contain the documentation prefixed with `--` in order for helm-docs to generate the documentation based on comments
+
+```yaml
+# -- storage class for PVCs associated with the apps
+storageClassName: gp2
+
+# -- node selector for event listener pod
+nodeSelector: ""
+#  ops: 'true'
+```
+
+### pre-commit
+
+This is needed to check/autofix for various errors in HELM charts (git precommit hooks)
+
+
+```sh
+git clone git@github.com:saritasa-nest/saritasa-devops-helm-charts.git
+cd saritasa-devops-helm-charts
+pip install pre-commit
+pre-commit install
+```
+
+## ci/cd
+
+- is implemented using github actions (see implementation in .github/workflows/release.yml)
+- commit into `feature/` or `fix/` prefixed branches in order for github actions to compile and release new version of the chart. See [this](https://github.com/saritasa-nest/saritasa-devops-helm-charts/blob/c8ef263796b0c59e703a6e1f4fb16b9c0738d180/.github/workflows/release.yml#L5-L8)
+- make sure Chart.yaml has minor version bumped, each time you want to release a new version
 
-For WebRTC chat
+After successfult build (see [Actions](https://github.com/saritasa-nest/saritasa-devops-helm-charts/actions)) please make sure to update local helm saritasa repository and see that the desired chart got updated (new chart version released.)
 
-## tekton
+If you want to see the result of the build (packaged HELM chart) you can do:
 
-tekton engine
+```sh
```

Comment:
As I understand, we have 2 typical cases for which would be good to have described in README.
1. What to do with erroneously released version (ex. I committed it, but found an error and want to "unrelease" a version and make another commit and publish it under same version. Probably, it is not a good idea to do exactly this way and better to release a next build, but IMHO this scenario worths mentioning.
2. I noticed, that new char version is released before mere to 'master' branch. On merge to master Action also starts build and fails, because the same version already exists. Maybe also worths to mention that it is "OK".

Reply:
1. so for 1 - you should remove the tag.
```sh
git tag -d demo-0.0.3 && git push origin :refs/tags/demo-0.0.3
```
2. yes, because we need to test it and argocd should download a tagged version of the HELM chart. Which requires the "release". 

So this is what I do:
- open `fix/` or `feature/` branch and perform changes
- test locally (`helm template .`), which is not applied inside kubernetes cluster.
- bump Chart.yaml file with a newer version (minor number increased)
- adjust argo-cd app template to use a new version from the step above
- test (if something is not working o to step 2 above)

Here is what we can do to simplify the naming:

# strategy 1:
- `fix` branch: x.y.Z (we allow only the 3rd digit to be bumped)
- `feature` branch: x.Y.0 (we allow bump of the 2nd digit)

# strategy 2:
- `fix` branch: x.y.Z-fix1, x.y.Z-fix23 (we allow only the 3rd digit to be bumped)
- `feature` branch: x.Y.0-alpha1, x.Y.0-alpha2 (we allow bump of the 2nd digit)
then once approved and merged into main bump in the main with
- `fix` branch: x.y.Z
- `feature` branch: x.Y.0

that way we know we're working off the main branch and not fix/feature branches.

Thoughts? @saritasa-nest/saritasa-devops

Reply:
I think `Strategy 1` is clear and sufficient in our case. `Strategy 2` would make sense if we would expect a lot of simultaneous changes and mandatory multiple rounds/stages of testing before release.

Reply:
@dmitry-mightydevops I don’t see how these strategies will help us. What are they for? What problem do they solve? As I understand it - if I create a feature/0.18.5 branch and send 2 commits there, the second commit will break at the ci stage.

Reply:
@SergeyDeminSaritasa just to streamline tagging experience using the gitflow approach.
the discussion is about the following:

a. you open a feature branch
b. you perform changes
c. you bump the CHART.yaml
d. you get feedback asking to perform changes. 

That's what the discussion is. How to tag the version when you perform updates after step "d".